TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py

```diff
@@ -0,0 +1,70 @@
+import subprocess
+import os
+import logging
+from typing import Any, Dict, List, Optional
+
+from console_link.models.command_result import CommandResult
+
+
+logger = logging.getLogger(__name__)
+
+FlagOnlyArgument = None
+
+
+class CommandRunner:
+    def __init__(self, command_root: str, command_args: Dict[str, Any], sensitive_fields: Optional[List[str]] = None,
+                 run_as_detatched: bool = False, log_file: Optional[str] = None):
+        self.command_args = command_args
+        self.command = [command_root]
+        for key, value in command_args.items():
+            self.command.append(key)
+            if value is not FlagOnlyArgument:
+                if type(value) is not str:
+                    value = str(value)
+                self.command.append(value)
+
+        self.sensitive_fields = sensitive_fields
+        self.run_as_detached = run_as_detatched
+        self.log_file = log_file
+
+    def run(self) -> CommandResult:
+        if self.run_as_detached:
+            return self._run_as_detached_process(self.log_file)
+        return self._run_as_synchronous_process()
+
+    def sanitized_command(self) -> List[str]:
+        if not self.sensitive_fields:
+            return self.command
+        display_command = self.command.copy()
+        for field in self.sensitive_fields:
+            if field in display_command:
+                field_index = display_command.index(field)
```

Comment:
nit: I'm still a bit hesitant on this logic and would rather see us use command_args for this. It's unlikely to cause a bug, but the possibility is still there to log something sensitive if another bug existed in the system.

E.g. 1:
if command_root == sensitive_field, there will be an issue (shouldn't happen)

e.g 2:
If value == sensitive field, there will be an issue (a lot less likely than before, but could still happen due to other issue.

Say someone accidentally sets their username as "--target-password", then password will be logged in plain text.